File: dayFive/java/Single number/SingleNumber.java
//javac -cp "junit-platform-console-standalone-1.11.0.jar:out" -d out SingleNumberTest.java SingleNumber.java
//java -cp "junit-platform-console-standalone-1.11.0.jar:out" org.junit.platform.console.ConsoleLauncher --scan-class-path

//every element appears twice except one 
// input nums = [2, 2, 1] output = 1
// input nums = [4, 1,2,1,2] output = 4

import java.util.Arrays;
public class SingleNumber{

    public static boolean isNotEmpty(int[] array){
        boolean lengthCheck = false;
        if(array.length > 0){
            lengthCheck = true;
        }
        else {
            lengthCheck = false;
        }
        return lengthCheck;
    }
    
    public static boolean isNotLessThanThreeElements(int[] array){
        boolean lengthCheck = false;
        if(array.length >= 3){
            lengthCheck = true;
        }
        else{
            lengthCheck = false;
        }
        return lengthCheck;
    }
    
    public static int getElementThatAppearsOnce(int[] array){
        
        int swap = 0;
        for(int index = 0; index < array.length; index++){
            for(int count = 0; count < array.length; count++){
                if(array[index] != array[count]){
                    swap = array[index];
                }
            }
        }
        return swap;
    }
}

File: dayFive/javaScript/singleNumber.js


function getElementThatAppearsOnce(array){
    let check = isNotEmpty(array);
    lengthCheck = isNotLessThanThreeElements(array);
    if(check === true && lengthCheck === true);
    
    let swap = 0;
    for(let index = 0; index < array.length; index++){
        for(let count = 0; count < array.length; count++){
            if(array[index] !== array[count]){
                swap = array[index];
            }
        }
    }
    return swap;
    
    
}
function isNotEmpty(array){
   let lengthCheck = false;
    if(array.length > 0){
        lengthCheck = true;
    }
    else if(array.length <= 0) {
        lengthCheck = false;
    }
    return lengthCheck;
}

function isNotLessThanThreeElements(array){
   let lengthCheck = false;
    if(array.length >= 3){
        lengthCheck = true;
    }
    else if (array.length < 3){
        lengthCheck = false;
    }
    return lengthCheck;
}
 let numbers = [2, 2, 1];
console.log(getElementThatAppearsOnce(numbers));
